File: frontend/src/types/segmentation.ts
// Mirrors backend/app/schemas/segmentation.py

export interface RfmPredictionRequest {
  recency: number;
  frequency: number;
  monetary: number;
}

export interface RfmPredictionResponse {
  cluster_id: number;
  segment_name: string;
  input: RfmPredictionRequest;
}

export interface RfmSegmentSummaryRow {
  Segment: string;
  Recency: number;
  Frequency: number;
  Monetary: number;
  customer_count: number;
}

export interface RfmSummary {
  cluster_label_map: Record<string, string>;
  segment_summary: RfmSegmentSummaryRow[];
  n_customers: number;
}

File: backend/app/schemas/segmentation.py
from __future__ import annotations

from pydantic import BaseModel, Field


class RfmPredictionRequest(BaseModel):
    recency: float = Field(..., ge=0, description="Days since the customer's last order")
    frequency: float = Field(..., ge=1, description="Number of distinct orders")
    monetary: float = Field(..., ge=0, description="Total spend across orders (BRL)")


class RfmPredictionResponse(BaseModel):
    cluster_id: int
    segment_name: str
    input: dict
